### PR TITLE
chore: Add a `[workspace.package]` section and refer to it from members

### PR DESCRIPTION
**Pull request by larseggert**
This means we only need to change stuff in one place from now on. Also eliminates the need for the `update_version` script, so remove that.

### CODE REVIEW

**Review by mxinden — APPROVED** [review]
:+1:

File: clippy.toml [hunk 1]
```diff
@@ -1 +0,0 @@
-msrv = "1.70.0"
```
Comment (martinthomson) [inline clippy.toml:1]: Just confirming: clippy will look at `workspace.package.rust-version` instead?  That's great.

**Review by martinthomson — APPROVED** [review]

File: clippy.toml (hunk 1 above)
Comment (larseggert) [reply clippy.toml:1]: It seems to, if there is a `rust-version.workspace = true` in the `Cargo.toml` of the member package.

https://doc.rust-lang.org/cargo/reference/workspaces.html#the-package-table

Comment (larseggert) [reply clippy.toml:1]: It seems to:

> [msrv](https://doc.rust-lang.org/clippy/lint_configuration.html#msrv)
> The minimum rust version that the project supports. Defaults to the rust-version field in Cargo.toml

https://doc.rust-lang.org/clippy/lint_configuration.html#msrv

**Comment by larseggert** [comment]
@martinthomson related question: Is there any point in updating `authors`, e.g., to `necko@mozilla.com`?

**Comment by martinthomson** [comment]
Yeah, it's probably time to do that.

**Comment by larseggert** [comment]
OK, I'll roll that into this one and land it.

**Pull request merged**